be/src/storage/tablet_manager.cpp

```diff
@@ -575,6 +575,11 @@ TabletSharedPtr TabletManager::find_best_tablet_to_compaction(CompactionType com
     for (int32_t i = tablet_shards_range.first; i < tablet_shards_range.second; i++) {
         std::shared_lock rlock(_tablets_shards[i].lock);
         for (auto [tablet_id, tablet_ptr] : _tablets_shards[i].tablet_map) {
+            if (tablet_id != tablet_ptr->tablet_id()) {
```

Comment:
Under which case, the two tablet_id id not equal?

Reply:
It happens, i don't know why so i add this log.